src/Text/Pandoc/Shared.hs

```diff
@@ -779,6 +780,31 @@ filterIpynbOutput mode = walk go
                     | otherwise = ""
         go x = x
 
+-- | Move formatting within code inlines to the outside,
+-- i.e. `Strong [ Code ("",[],[]) "Strong code" ]` (issue #7525)
+formatCode :: Inlines -> Inlines
+formatCode =
+  B.fromList
+    . concatMap fmt
+    . groupBy (\a b -> isPlaintext a && isPlaintext b)
+    . B.toList
+  where
+    isPlaintext (Str _) = True
+    isPlaintext Space = True
+    isPlaintext SoftBreak = True
+    isPlaintext (Quoted _ _) = True
+    isPlaintext _ = False
+    fmt xs
+        | all isPlaintext xs = B.toList $ B.code $ stringify xs
+        | any isPlaintext xs = B.toList $ formatCode $ B.fromList xs
```

Comment:
Under what condition could we have `any isPlaintext xs` but not `all isPlaintext xs`?

Reply:
This is for something with multiple inlines, like

```latex
\texttt{\sout{Normal. \underline{Underlined.}}}
```

While walking through this expression, there will be a point where one of the inlines in `\sout` is handled, but not the other:

```
Strikeout [ Str "Normal. ", Underline [ Code ("",[],[]) "Underlined." ] ]
```

Reply:
I see. Instead, why not at line 803 have `concatMap fmt . groupBy (\a b -> isPlaintext a && isPlaintext b)`?
Then we could be confident that `fmt` would never apply to mixed lists.  That might make things a bit clearer.

Reply:
Agreed, that looks nicer. I've extracted the `concatMap fmt . groupBy (\a b -> isPlaintext a && isPlaintext b)` into a separate function. Please let me know if you can think of a better name for it (or if I should leave it inlined).